go-controller/pkg/factory/factory.go

```diff
@@ -687,6 +719,9 @@ func NewClusterManagerWatchFactory(ovnClientset *util.OVNClusterManagerClientset
 	if err := ipamclaimsapi.AddToScheme(ipamclaimsscheme.Scheme); err != nil {
 		return nil, err
 	}
+	if err := nadapi.AddToScheme(nadscheme.Scheme); err != nil {
```

Comment:
Can we make this condition tool with  `if util.IsNetworkSegmentationSupportEnabled() {` ?

Reply:
I don't see the other features doing this, besides given its costless I think its fine to let it be...